packages/core/types/src/file/providers/s3.ts

```diff
@@ -1,7 +1,8 @@
 export interface S3FileServiceOptions {
   file_url: string
-  access_key_id: string
-  secret_access_key: string
+  access_key_id?: string
+  secret_access_key?: string
+  authentication_method?: "access-key" | "s3-iam-role"
```

Comment:
One open point we had with @sgirones is whether deciding the auth method should be an explicit action (it defaults to access-key), or implicit by the fact that the access key ID and secret access key are not passed.

WDYT?

Reply:
The current approach makes sense to me – enforcing a clear intent for auth seems reasonable

Reply:
for more context: the AWS SDKs usually follow the standard configuration layers, the most common being env vars.

Usually, you would do something like: `AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY npm run start`

In this specific case, instead of `s3-iam-role` we should have something like `from-env`, otherwise the user may expect to be able to set the iam role through the config.